### PR TITLE
Added a test to verify the instrumentation support for the Spring HTTP client RestClient.

### LINKED ISSUE #3441: Spring Framework 6.1 - Add support for RestClient
Add support for `org.springframework.web.client.RestClient`.

### PR DESCRIPTION
**Pull request by videnkz**
## What does this PR do?
Added a test to verify the instrumentation support for the Spring HTTP client `RestClient`.

closes #3441 

## Checklist

- [x] This is something else
  - [x] I have added tests
  - [x] ~~I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)~~
  - [x] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/docs/supported-technologies.asciidoc)

### CODE REVIEW

File: apm-agent-plugins/apm-spring-resttemplate/apm-spring-restclient-test/src/test/java/co/elastic/apm/agent/restclient/SprintRestClientIntegration.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.junit.Test;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.client.RestTemplate;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SprintRestClientIntegration extends AbstractInstrumentationTest {
```
[the remark this answers is not in the document]
Comment (videnkz) [reply apm-agent-plugins/apm-spring-resttemplate/apm-spring-restclient-test/src/test/java/co/elastic/apm/agent/restclient/SprintRestClientIntegration.java]: removed

**Comment by JonasKunz** [comment]
@elasticmachine run elasticsearch-ci/docs

**Review by JonasKunz — DISMISSED** [review]
Thanks for implementing this!

**Review by jackshirazi — APPROVED** [review]

**Comment by JonasKunz** [comment]
@elasticmachine run elasticsearch-ci/docs

**Review by JonasKunz — APPROVED** [review]

**Pull request merged**